pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -331,4 +344,31 @@ pub mod pallet {
 			}
 		}
 	}
+
+	impl AssociateAccountRequest {
+		pub fn verify<T: Config>(
+			&self,
+			did_identifier: <T as Config>::DidIdentifier,
+			expiration: <T as frame_system::Config>::BlockNumber,
+		) -> bool {
+			let encoded_payload = (&did_identifier, expiration).encode();
+			match self {
+				AssociateAccountRequest::Dotsama(acc, proof) => proof.verify(
+					&get_wrapped_payload(&encoded_payload[..], crate::signature::WrapType::Substrate)[..],
+					acc,
+				),
+				AssociateAccountRequest::Ethereum(acc, proof) => proof.verify(
+					&get_wrapped_payload(&encoded_payload[..], crate::signature::WrapType::Ethereum)[..],
+					acc,
+				),
+			}
+		}
+
+		pub fn get_linkable_account(&self) -> LinkableAccountId {
+			match self {
+				AssociateAccountRequest::Dotsama(acc, _) => LinkableAccountId::AccountId32(acc.clone()),
+				AssociateAccountRequest::Ethereum(acc, _) => LinkableAccountId::AccountId20(*acc),
+			}
+		}
+	}
```

Comment:
i like to keep the impl block and the struct close together. Maybe even move them to their own file.